src/code/reducers/modal-dialog.js

```diff
@@ -18,15 +18,19 @@ const defaultRightButton = {
 export default function modalDialog(state = initialState, action) {
   switch (action.type) {
     case actionTypes.MODAL_DIALOG_SHOWN:
-      return state.merge({
-        show: true,
-        message: action.message,
-        explanation: action.explanation,
-        rightButton: action.rightButton || defaultRightButton,
-        leftButton: action.leftButton,
-        showAward: action.showAward,
-        top: action.top
-      });
+      if (action.showAward || (action.rightButton && action.rightButton.action === "advanceTrial")) {
```

Comment:
Again, this is a little bit of a hack so that the modal dialog doesn't handle messages that `notifications` are listening for. The actions should be reworked